src/navigators/RootStack.tsx

```diff
@@ -49,28 +49,31 @@ export function RootStack() {
     const Stack = createNativeStackNavigator<LoggedInRootParamList>();
     const hasAccountAndProject = !!(activeProject?.id && account?.id);
 
-    const initialRoute = !hasAccountAndProject
-      ? 'InviteRequired'
-      : shouldRenderConsentScreen
-      ? 'screens/ConsentScreen'
-      : 'app';
-
-    if (loadingAccountOrProject && initialRoute === 'InviteRequired') {
+    if (loadingAccountOrProject) {
       return (
         <ActivityIndicatorView
-          message={t('root-stack-waiting-for-account-and-project', 'Loading')}
+          message={t(
+            'root-stack-waiting-for-account-and-project',
+            'Loading account',
+          )}
         />
       );
     }
 
-    if (loadingConsents && initialRoute === 'screens/ConsentScreen') {
+    if (loadingConsents) {
       return (
         <ActivityIndicatorView
-          message={t('root-stack-waiting-for-consents', 'Loading')}
+          message={t('root-stack-waiting-for-consents', 'Loading consents')}
         />
       );
     }
 
+    const initialRoute = !hasAccountAndProject
+      ? 'InviteRequired'
+      : shouldRenderConsentScreen
+      ? 'screens/ConsentScreen'
+      : 'app';
```

Comment:
An important trick with this `initialRouteName` approach is that you can't really change it once the `return` below has been rendered.  So after a recent refactor, we get `shouldRenderConsentScreen` initially coming through here as `false` and then later as `true`.  Unless we avoid rending the `Stack.Navigator` below on those fully loading (accounts & consents), the `initialRouteName` doesn't behave/react like we need it to.  The result in this case is I'm consistently seeing consents in a proposed state that aren't being shown to the user, without this PR.